reporter/sources/php/exceptions.py

```diff
@@ -0,0 +1,101 @@
+import json
+
+from reporter.helpers import is_production_host
+from reporter.reports import Report
+
+from common import PHPLogsSource
+
+
+class PHPExceptionsSource(PHPLogsSource):
+    """
+    Report errors and exceptions logged via WikiaLogger
+    """
+    REPORT_LABEL = 'PHPExceptions'
+
+    FULL_MESSAGE_TEMPLATE = """
+h1. {exception}
+
+{message}
+
+h5. Backtrace
+{backtrace}
+"""
+
+    LIMIT = 50000
+
+    def _get_entries(self, query):
+        """ Return errors and exceptions reported via WikiaLogger with error severity """
+        return self._kibana.query_by_string(
+            query='severity:"^error" AND program: "apache2" AND -@exception.class: "DBConnectionError"',
+            limit=self.LIMIT
+        )
+
+    def _filter(self, entry):
+        # filter out by host
+        # "@source_host": "ap-s10",
+        host = entry.get('@source_host', '')
+        if not is_production_host(host):
+            return False
+
+        return True
+
+    def _normalize(self, entry):
+        """ Normalize using the exception class and message """
+        exception = entry.get('@exception', {})
+        exception_class = exception.get('class')
+        env = self._get_env_from_entry(entry)
+
+        message = entry.get('@message', '')
+
+        # use a message from the exception
+        if exception_class == 'WikiaException':
+            message = exception.get('message')
+
+        entry['@normalized_message'] = message
+
+        return '{}-{}-{}'.format(env, exception_class or 'None', message)
+
+    def _get_report(self, entry):
```

Comment:
The function is pretty lengthy, would it make sense to move parts to subroutines i.e. generate_description etc. So that each function fits on one screen.